### PR TITLE
fix. toolbar disable,view comment not remove

### LINKED ISSUE #3002: Bug: Filters 'except' option not removing DebugToolbar comment for view
**Describe the bug**
[`except`](https://codeigniter4.github.io/userguide/incoming/filters.html?highlight=except#globals) option in filters disables the [Debug Toolbar](https://codeigniter4.github.io/userguide/testing/debugging.html?highlight=toolbar#the-debug-toolbar) but adds the comment `<!-- DEBUG-VIEW START -->`. This comment breaks the css when I m trying to load the view as `CSS` with `$this->response->setContentType('text/css')`. May causes the problem for other content types also.

**CodeIgniter 4 version**
v4.0.3

**Affected module(s)**
CodeIgniter\View

**Expected behavior, and steps to reproduce if appropriate**

Expected that the comment should not be added when `URI` available in `except` filter.

**Filters.php**
```
public $globals = [
	'before' => [],
	'after'  => [
		'toolbar' => ['except' => 'style'],
		//'honeypot'
	],
];
```
https://github.com/codeigniter4/CodeIgniter4/blob/9bbc99136dce5b57baf8d27064ce85b9c2ee619c/system/View/View.php#L262-L282

**Context**
 - OS: Windows 7
 - Web server: Apache/2.4.41
 - PHP version: 7.3.9

### PR DESCRIPTION
**Pull request by Instrye**
- fix #3002. toolbar disable,but view comment not remove.
- ~~delete filter argumets support. because, filter muset `implements FilterInterface`. `before` only receive request. so, if filter argumetns send to `before`, will get an error.~~
- Alias resolution is no longer done in the `run` method, `Filters::$filtersClass` save alias to class
**Checklist:**
- FiltersTest. append `aliases` to config
- [x] Securely signed commits
- [x] Component(s) with PHPdocs
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by MGatner** [comment]
Thank you @Instrye - this looks like quite a bit of work and some good changes. I don't have my head around it, I'll have to try another day. In the meantime, anybody else willing to give it a review?

**Comment by paulbalandan** [comment]
This needs to be revised to incorporate changes in `FilterInterface`.

**Comment by samsonasik** [comment]
@MGatner could you re-check it?

File: system/View/View.php [hunk 1]
```diff
@@ -269,7 +269,7 @@ public function render(string $view, array $options = null, bool $saveData = nul
 
 		$this->logPerformance($this->renderVars['start'], microtime(true), $this->renderVars['view']);
 
-		if ($this->debug && (! isset($options['debug']) || $options['debug'] === true))
+		if (($this->debug && (! isset($options['debug']) || $options['debug'] === true)) && in_array('CodeIgniter\Filters\DebugToolbar', service('filters')->getFiltersClass()['after']))
```
Comment (samsonasik) [inline system/View/View.php:272]: can use strict in_array check: 

```suggestion
              if (($this->debug && (! isset($options['debug']) || $options['debug'] === true)) && in_array('CodeIgniter\Filters\DebugToolbar', service('filters')->getFiltersClass()['after'], true))
```

File: tests/system/Filters/FiltersTest.php [hunk 2]
```diff
@@ -913,4 +998,33 @@ public function testFilterAlitasMultiple()
 		$this->assertEquals('http://exampleMultipleCSP.com', $request->csp);
 	}
 
+	public function testFilterClass()
+	{
+		$config  = [
+			'aliases' => [
+				'multipeTest' => [
```
Comment (MGatner) [inline tests/system/Filters/FiltersTest.php:1005]:
```suggestion
				'multipleTest' => [
```

File: tests/system/Filters/FiltersTest.php [hunk 3]
```diff
@@ -913,4 +998,33 @@ public function testFilterAlitasMultiple()
...
+		$config  = [
+			'aliases' => [
+				'multipleTest' => [
+					'CodeIgniter\Filters\fixtures\Multiple1',
+					'CodeIgniter\Filters\fixtures\Multiple2',
+				],
+			],
+			'globals' => [
+				'after' => [
+					'multipeTest',
```
Comment (MGatner) [inline tests/system/Filters/FiltersTest.php:1012]:
```suggestion
					'multipleTest',
```

**Comment by MGatner** [comment]
A few issues failing the static analysis check, but otherwise this looks good. Waiting on Travis. @samsonasik should we merge despite the CS failures? Or resolve those first?

**Comment by samsonasik** [comment]
@MGatner I think code static analysis check need to incorporated first: 

```
 ------ ----------------------------------------------------------------------- 
  Line   system/Filters/Filters.php                                             
 ------ ----------------------------------------------------------------------- 
  157    Access to an undefined property                                        
         CodeIgniter\Filters\Filters::$argumentsClass.                          
  183    Access to an undefined property                                        
         CodeIgniter\Filters\Filters::$argumentsClass.                          
  338    Access to an undefined property                                        
         CodeIgniter\Filters\Filters::$argumentsClass.                          
  495    Return typehint of method                                              
         CodeIgniter\Filters\Filters::processAliasesToClass() has invalid type  
         CodeIgniter\Filters\type.                                              
  497    Method CodeIgniter\Filters\Filters::processAliasesToClass() should     
         return CodeIgniter\Filters\type but return statement is missing.      
```

File: system/Filters/Filters.php [hunk 4]
```diff
@@ -60,6 +60,11 @@ class Filters
 		'after'  => [],
 	];
 
+	protected $filtersClass = [
+		'before' => [],
+		'after'  => [],
+	];
+
```
Comment (samsonasik) [inline system/Filters/Filters.php:67]: new property can be added:

```suggestion

           /** @var array */
          protected $argumentsClass;

```

File: system/Filters/Filters.php [hunk 5]
```diff
@@ -474,6 +477,36 @@ protected function processFilters(string $uri = null)
 		}
 	}
 
+	/**
+	 * filter alias to class
+	 *
+	 * @return type
```
Comment (samsonasik) [inline system/Filters/Filters.php:483]: docblock can be updated to:

```suggestion
       * @return void
       * @throws FilterException
```

**Review by michalsn — CHANGES_REQUESTED** [review]

**Comment by MGatner** [comment]
@Instrye It looks like one more round of tweaks, if you're available for those.

**Comment by Instrye** [comment]
😂I want to provide quality phpdoc, But my English level hinders me.

File: system/Filters/Filters.php [hunk 6]
```diff
@@ -334,16 +343,29 @@ public function enableFilter(string $name, string $when = 'before')
...
 		}
 
+		if (is_array($this->config->aliases[$name]))
+		{
+			$classNames = $this->config->aliases[$name];
+		}
+		else
+		{
+			$classNames = [$this->config->aliases[$name]];
+		}
```
[the remark this answers is not in the document]
Comment (Instrye) [reply system/Filters/Filters.php]: thanks

**Review by michalsn — APPROVED** [review]
Looks good, thanks! 

You only have to resolve some conflicts with the develop branch and I think we're good to go.

**Comment by Instrye** [comment]
@michalsn Everything's ready.

**Comment by michalsn** [comment]
Thanks!

**Pull request merged**